### DISCUSSION TITLE
[BUG] `Can't reactivate license`

### DISCUSSION THREAD

**Discussion by JFuqX** [dashingsoft/pyarmor#1347] [Q&A] [answered]
Error on windows when upgrading from ``pyarmor-regfile-xxxx.zip`` it was an upgraded license

```
6116 MainProcess 2023-06-28 00:23:29,737
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\__main__.py", line 706, in main
    main_entry(sys.argv[1:])
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\__main__.py", line 694, in main_entry
    args.func(ctx, args)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\__main__.py", line 305, in cmd_reg
    reg.register_regfile(regfile)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\register.py", line 169, in register_regfile
    self.update_token()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\register.py", line 112, in update_token
    Pytransform3._update_token(self.ctx)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\core\__init__.py", line 127, in _update_token
    m = Pytransform3.init(ctx)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python39\lib\site-packages\pyarmor\cli\core\__init__.py", line 97, in init
    m.init_ctx(ctx)
RuntimeError

```

**Comment by jondy** [comment]
How do you upgrade the old license?

**Comment by JFuqX** [comment]
I did it regularly had an old license (v7) had an upgrade license, so I upgraded it on my macOS, then I got the ``pyarmor-regfile-xxxx.zip`` which I was told to keep to do the activation on other devices because both codes (old activation v7 code and the upgrade code) don't work anymore.

**Comment by jondy** [comment]
What's your old license no and what's the `xxxx` in the `pyarmor-regfile-xxxx.zip`?

**Comment by JFuqX** [comment]
My old license was also 1977

![Bildschirmfoto 2023-06-29 um 02 12 49](https://github.com/dashingsoft/pyarmor/assets/56825613/e3a78902-d6e2-4fe3-970b-a6e4448750b7)

**Comment by jondy** [comment]
Please first upgrade pyarmor version to latest 8.2.8

You should report the issue following the template, no version or necessary information, it make something complex.

**Comment by JFuqX** [comment]
It is already upgraded to the newest version and I provided you all information I have.

**Comment by JFuqX** [comment]
```
$ pyarmor reg pyarmor-regfile-1977.zip
INFO     Python 3.9.8
INFO     Pyarmor 8.2.8 (trial), 000000, non-profits
INFO     Platform windows.x86_64
```

It should be a pyarmor pro license but its not possible to register the license under pyarmor8

**Comment by jondy** [comment]
Is there any fireware to block the license server? I have found tokens of this license has been created in the server side, but the client side got nothing. It most likely network data is blocked.

**Comment by jondy** [comment]
And what's the error message? There is only `RuntimeError`, is there any extra error message?

**Comment by JFuqX** [comment]
sadly not I provided you everything I know ...

**Comment by JFuqX** [comment]
C:\Users\Administrator\Desktop>pyarmor-7 register pyarmor-regfile-1977.zip
INFO     PyArmor Version 8.2.9
INFO     Python 3.9.8
INFO     Start to register keyfile: pyarmor-regfile-1977.zip
INFO     Save registration data to: C:\Users\Administrator\.pyarmor
INFO     Extracting license.lic
INFO     Extracting .pyarmor_capsule.zip
INFO     This keyfile has been registered successfully.

C:\Users\Administrator\Desktop>pyarmor --version
Pyarmor 8.2.9 (trial), 000000, non-profits

License Type    : pyarmor-trial
License No.     : pyarmor-vax-000000
License To      :
License Product : non-profits

BCC Mode        : No
RFT Mode        : No

Notes
* Can't obfuscate big script and mix str


what's weird is that it works when I use the pyarmor7 command to upgrade it, but when I use the pyarmor 8 command to check the version it doesn't work ....

**Comment by jondy** [comment, marked as answer]
I guess that firmware in your side blocks the data from pyarmor server side, maybe your machine is not direct to connect to internet, for example, in company network or in a school network. 

You can do a test in a new network environment.

**Reply by JFuqX** [reply]
It is installed on a Windows Remote server so if there would be any issues I couldn't connect to the server.

If there would be an issue from my side my previous message wouldn't make any sense.

**Reply by jondy** [reply]
Could you check Windows Defender Settings on Windows Remote server?

And also do a test in another local machine.

**Reply by JFuqX** [reply]
Sadly there is no fix can you please provide me a new license so I can actually make use of my paid commercial upgrade?
